### PR TITLE
qa: s3tests support for absolute & subdomain calling format 

### PR DESCRIPTION
**Pull request by theanalyst**
Rework of https://github.com/ceph/ceph/pull/15063 atop of https://github.com/ceph/ceph/pull/15071, adds s3tests for absolute urls & subdomain calling format, since the s3tests pr https://github.com/ceph/s3-tests/pull/162 is not yet merged, added an option to specify a git_remote, for absolute urls to pass https://github.com/ceph/ceph/pull/12861 needs to go in too.

### CODE REVIEW

**Comment by theanalyst** [comment]
simple qa run: http://pulpito.ceph.com/abhi-2017-05-16_14:11:24-rgw-wip-absolute-urls-rebase---basic-smithi/

**Comment by theanalyst** [comment]
jenkins test this please

**Comment by tchaikov** [comment]
@theanalyst ping?

**Comment by liewegas** [comment]
@theanalyst ping

File: qa/tasks/dnsmasq.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    remote.run(args=['sudo', 'systemctl', 'restart', 'dnsmasq'])
+    remote.run(args=['sudo', 'systemctl', 'status', 'dnsmasq'])
+    # verify dns name is set
+    remote.run(args=['ping', '-c', '4', cnames.keys()[0]])
+
+    yield
+
+    log.info('Removing dnsmasq configuration from remote %s..', remote.name)
+    # restore resolv.conf
+    misc.sudo_write_file(remote, '/etc/resolv.conf', resolv_conf)
```
Comment (cbodley) [inline qa/tasks/dnsmasq.py]: something seems to have changed, and this line no longer works for the hadoop-s3a tests:
> 2017-12-23T17:32:22.183 INFO:teuthology.orchestra.run.ovh080:Running: 'sudo python -c \'import shutil, sys; shutil.copyfileobj(sys.stdin, file(sys.argv[1], "wb"))\' /etc/resolv.conf'
2017-12-23T17:32:22.396 INFO:teuthology.orchestra.run.ovh080.stderr:Traceback (most recent call last):
2017-12-23T17:32:22.396 INFO:teuthology.orchestra.run.ovh080.stderr:  File "<string>", line 1, in <module>
2017-12-23T17:32:22.397 INFO:teuthology.orchestra.run.ovh080.stderr:IOError: [Errno 13] Permission denied: '/etc/resolv.conf'

Comment (theanalyst) [reply qa/tasks/dnsmasq.py]: I'll rebase again, this was old when https://github.com/ceph/ceph/pull/15071 wasn't merged yet

**Comment by mattbenjamin** [comment]
@cbodley @theanalyst is this still active?

**Comment by cbodley** [comment]
yep

**Pull request closed without merging**